src/ra/groupavg.h

```diff
@@ -0,0 +1,85 @@
+#ifndef RA_GroupAvg_H_
+#define RA_GroupAvg_H_
+
+#include <type_traits>
+#include <utility>
+
+#include <iostream>
+
+#include "range/v3/all.hpp"
+
+/* This operator should only be used following ra::groupby */
+
+namespace fluent {
+namespace ra {
+
+template <typename PhysicalChild, std::size_t Is>
+class PhysicalGroupAvg {
+
+  typedef ranges::iterator_value_t<ranges::range_iterator_t<typename std::result_of_t<decltype(&PhysicalChild::ToRange)(PhysicalChild)>>> rngtype;
+  typedef ranges::iterator_value_t<ranges::range_iterator_t<rngtype>> valuetype;
+  //typedef typename std::tuple_element<Is, valuetype>::type keytype;
+
+ public:
+  PhysicalGroupAvg(PhysicalChild child) : child_(std::move(child)) {}
+
+  auto ToRange() {
+    return child_.ToRange() | ranges::view::transform([](const auto& rng) {
+          auto length = ranges::size(rng);
+          std::vector<valuetype> v = rng | ranges::to_<std::vector<valuetype>>();
+          decltype(std::get<Is>(std::declval<valuetype>())) s = 0;
+          for (auto& t: v)
+            s += std::get<Is>(t);
+          return std::make_tuple(s/length);
```

Comment:
Shouldn't a group by followed by an average return the grouped by columns in addition to the average? Also what if length is 0?